### PR TITLE
Ls cleanup

### LINKED ISSUE #739: Standardise command descriptions for output of "ls"
At the moment, some are missing, others are too long (wrapping is pretty disruptive), and many are uninformative or verbose.

We can fix all of these, and for scripts the first two can be added to the docs linter to ensure they stay fixed.

This requirement should also be noted in the `Contributing` doc.

### PR DESCRIPTION
**Pull request by PeridexisErrant**
Closes #739 

>At the moment, some are missing, others are too long (wrapping is pretty disruptive), and many are uninformative or verbose.

>We can fix all of these, and for scripts the first two can be added to the docs linter to ensure they stay fixed.  This requirement should also be noted in the Contributing doc.

Done.  It turns out that some existing help wasn't being picked up.  I fixed that, and also where I saw it added useful help instead of just the script name.

### CODE REVIEW

File: plugins/reveal.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ command_result nopause(color_ostream &out, vector<string> & params);
 
 DFhackCExport command_result plugin_init ( color_ostream &out, vector <PluginCommand> &commands)
 {
-    commands.push_back(PluginCommand("reveal","Reveal the map. 'reveal hell' will also reveal hell. 'reveal demon' won't pause.",reveal,false,
+    commands.push_back(PluginCommand("reveal","Reveal the map. Args 'hell', 'demon' are risky.",reveal,false,
```
Comment (lethosor) [inline plugins/reveal.cpp]: You probably don't need to list arguments here since they're listed in the help text below.

**Comment by lethosor** [comment]
What's a script plugin?
It would be better to stop `ls` from requiring a space (and the travis check as well), since there's really no reason to.

File: plugins/devel/memview.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@ DFHACK_PLUGIN("memview");
 
 DFhackCExport command_result plugin_init (color_ostream &out, std::vector <PluginCommand> &commands)
 {
-    commands.push_back(PluginCommand("memview","Shows memory in real time. Params: adrr length refresh_rate. If addr==0 then stop viewing",memview));
+    commands.push_back(PluginCommand("memview","Shows DF memory in real time.",memview));
```
Comment (lethosor) [inline plugins/devel/memview.cpp]: It would be nice to add extended help text here.

File: scripts/autofarm.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+# Select cropt to plant based on current stocks
```
Comment (dscorbett) [inline scripts/autofarm.rb]: “cropt” → “crops”

**Comment by lethosor** [comment]
2882422 should fix the space issue.

**Comment by PeridexisErrant** [comment]
"script plugin" -> I have no idea why I typed that.  Your patch to `ls` is clearly better; I'm just read-only in CPP and don't know the DFHack code that well.  Sounds like it's done now?

**Comment by expwnent** [comment]
Yeah, looks good. For `ls` the description should be short as long as there's a longer more detailed explanation available. If we include the docs offline (issue #801)  then we should consider a way of automatically opening them to the relevant part with the help command. Maybe we should do that anyway except directly open the online version in the default browser. I don't know how to to do that but it would be neat.

**Comment by PeridexisErrant** [comment]
Note:  it will be far more difficult to merge this if it's left for after #723 (sorry, @lethosor, I keep finding more dependent issues!)

> If we include the docs offline (issue #801) then we should consider a way of automatically opening them to the relevant part with the help command. Maybe we should do that anyway except directly open the online version in the default browser.

There's some similar discussion in #869; I proposed `search.lua` which would just open the docs search page with any arguments as the url query parameter (properly escaped, of course).  Eg `search plugin script` would open a browser to https://dfhack.readthedocs.org/en/stable/search.html?q=plugin+script - or the local copy is available.

**Comment by expwnent** [comment]
I can't conveniently merge it right now so I just marked it for the next release milestone.

**Comment by PeridexisErrant** [comment]
@lethosor @expwnent - could this be merged before r1?  If it would help, I'm happy to rebase for easier merging.

File: scripts/make-monarch.lua [hunk 4]
```diff
@@ -1,4 +1,4 @@
---set target unit as king/queen
+-- set target unit as king/queen
```
Comment (lethosor) [inline scripts/make-monarch.lua]: Isn't necessary anymore

**Comment by lethosor** [comment]
I'll probably be able to tomorrow. Any idea how bad the merge conflict is?

**Comment by PeridexisErrant** [comment]
Pull rebased on develop, and dropped unneeded fixes.  Should be ready and easy to merge!

Also a nice education in the power of `git rebase -i`, wow.  Took a few attempts, but it'll be a lot of fun if I need that again :smile:

**Pull request merged**